backend/src/apiserver/server/pipeline_server.go

```diff
@@ -98,427 +108,1027 @@ type PipelineServer struct {
 	}
 
-	pipelines, total_size, nextPageToken, err := s.resourceManager.ListPipelines(filterContext, opts)
+	// Check authorization
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Verb: common.RbacResourceVerbDelete,
+	}
+	err := s.canAccessPipeline(ctx, pipelineId, resourceAttributes)
 	if err != nil {
-		return nil, util.Wrap(err, "List pipelines failed.")
+		return util.Wrap(err, fmt.Sprintf("[PipelineServer %s]: Failed to delete a pipeline due authorization error for pipeline id %v.", s.options.ApiVersion, pipelineId))
 	}
-	apiPipelines := ToApiPipelines(pipelines)
-	return &api.ListPipelinesResponse{Pipelines: apiPipelines, TotalSize: int32(total_size), NextPageToken: nextPageToken}, nil
+
+	return s.resourceManager.DeletePipeline(pipelineId)
 }
 
-func (s *PipelineServer) DeletePipelineV1(ctx context.Context, request *api.DeletePipelineRequest) (*empty.Empty, error) {
-	if s.options.CollectMetrics {
-		deletePipelineRequests.Inc()
+// Removes a model.PipelineVersion.
+func (s *PipelineServer) deletePipelineVersion(ctx context.Context, pipelineId string, pipelineVersionId string) error {
+	// Fail fast
+	if pipelineId == "" {
+		return util.NewInvalidInputError("[PipelineServer %s]: Failed to delete a pipeline version id %v due missing pipeline id.", s.options.ApiVersion, pipelineVersionId)
 	}
-	resourceAttributes := &authorizationv1.ResourceAttributes{
-		Verb: common.RbacResourceVerbDelete,
+	if pipelineVersionId == "" {
+		return util.NewInvalidInputError("[PipelineServer %s]: Failed to delete a pipeline version due missing pipeline version id.", s.options.ApiVersion)
 	}
-	err := s.CanAccessPipeline(ctx, request.Id, resourceAttributes)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to authorize the requests.")
+
+	// Check authorization
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Verb: common.RbacResourceVerbList,
 	}
-	err = s.resourceManager.DeletePipeline(request.Id)
+	err := s.canAccessPipeline(ctx, pipelineId, resourceAttributes)
 	if err != nil {
-		return nil, util.Wrap(err, "Delete pipelines failed.")
-	}
-	if s.options.CollectMetrics {
-		pipelineCount.Dec()
+		return util.Wrap(err, fmt.Sprintf("[PipelineServer %s]: Failed to delete a pipeline version id %v due to authorization error for pipeline id %v.", s.options.ApiVersion, pipelineVersionId, pipelineId))
 	}
 
-	return &empty.Empty{}, nil
+	return s.resourceManager.DeletePipelineVersion(pipelineVersionId)
 }
 
-func (s *PipelineServer) GetTemplate(ctx context.Context, request *api.GetTemplateRequest) (*api.GetTemplateResponse, error) {
-	resourceAttributes := &authorizationv1.ResourceAttributes{
-		Verb: common.RbacResourceVerbList,
+// v2beta1: creates a pipeline.
+// v1beta1: creates a pipeline and a pipeline version.
+func (s *PipelineServer) CreatePipelineCommon(ctx context.Context, r interface{}) (interface{}, error) {
```

Comment:
What's the purpose of this function, does it match to any api call?

Reply:
Agree. Removed.